netmanager/src/views/components/d3/CustomisableChart.js

```diff
@@ -170,12 +171,8 @@ const CustomisableChart = (props) => {
     if (isCustomPeriod(period)) {
       return [new Date(period.startDate), new Date(period.endDate)];
     }
-    let endDate = period.endDate ? new Date(period.endDate) : new Date();
-    let startDate = new Date(
-      endDate.getFullYear(),
-      endDate.getMonth(),
-      endDate.getDate() - period.unitValue
-    );
+    let endDate = new Date();
+    let startDate = moment().subtract(period.unitValue, 'days');
```

Comment:
@Codebmk Please set the unit from period object (period.unit). The unit could be months and days.